### PR TITLE
Using WarningsLibrary to query for warnings

### LINKED ISSUE #6616: A simple vector with warnings does not highlight the node
### Discord username

_No response_

### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

Screenshot from 2023-05-09 14-15-32

The above screenshot visualizes warnings, not data. So they clearly are there.

### Expected behaviour

Should be wrapped in orange, similar to our regular warnings.

Screenshot from 2023-05-09 14-55-22

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Enso Version

develop

### Browser or standalone distribution

Chrome

### Browser Version or standalone distribution

110.0.5481.100

### Operating System

Linux

### Operating System Version

_No response_

### Hardware you are using

_No response_

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Fixes #6616 by using `WarningsLibrary`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
- All code has been tested:
  - [x] Unit tests updated.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
obrazek

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala [hunk 1]
```diff
@@ -351,27 +351,32 @@ object ProgramExecutionSupport {
...
-                  warningsCount,
-                  warning,
-                  withWarnings.isLimitReached()
+        case _ =>
+          if (WarningsLibrary.getUncached().hasWarnings(value.getValue)) {
+            val warnings =
+              WarningsLibrary.getUncached().getWarnings(value.getValue, null)
+            val warningsCount = warnings.length
+            val warning =
+              if (warningsCount >= 1) {
```
Comment (hubertp) [inline engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala]: This changes the semantics. We only want to include the warning if it is a single one.

**Review by hubertp — CHANGES_REQUESTED** [review]

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala]: OK. Semantics is back in ba86adf - however I have to admit, I don't understand the reasoning - why do we want just a single warning?

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala [hunk 2]
```diff
@@ -351,27 +351,32 @@ object ProgramExecutionSupport {
...
-            }
-          Api.ExpressionUpdate.Payload.Value(
-            Some(
-              Api.ExpressionUpdate.Payload.Value
-                .Warnings(
-                  warningsCount,
-                  warning,
-                  withWarnings.isLimitReached()
+        case _ =>
+          if (WarningsLibrary.getUncached().hasWarnings(value.getValue)) {
```
Comment (Akirathan) [inline engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala]: minor: `getUncached()` should be used as `getUncached` - linter tells me a warning: "Java accessor method called with empty argument clause"

**Review by Akirathan — APPROVED** [review]
Good. Just minor nits. I hope the performance won't degrade by using uncached library? I guess we don't have any other option though.

**Review by hubertp — APPROVED** [review]

**Comment by JaroslavTulach** [comment]
> I hope the performance won't degrade by using uncached library? I guess we don't have any other option though.

This is not PE code. We have to use uncached version or we would need to do a more complicated rewrite.

**Comment by JaroslavTulach** [comment]
There is a [failure in RuntimeServerTest](https://github.com/enso-org/enso/actions/runs/5014017709/jobs/8987786956?pr=6751#step:10:5545) - should the expected message just be updated, @4e6? The test seems to trigger the new code which then sends:
```scala
Api.ExpressionUpdate.Payload.Value(
    Some(
      Api.ExpressionUpdate.Payload.Value
        .Warnings(
          warningsCount,
          warning,
          WarningsLibrary.getUncached().isLimitReached(value.getValue)
        )
    )
)
```
even when the `value.getValue` is `Vector` - that's desirable change in behavior, I think. Updated in 123f65e793

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala (hunk 2 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala]: My IDE is silent, but here you are: 36d114d2c4

**Comment by JaroslavTulach** [comment]
Failure in [RuntimeVisualizationsTest](https://github.com/enso-org/enso/actions/runs/5021165279/jobs/9006015107?pr=6751#step:10:5572) fixed in 224ae5c4e1

**Pull request merged**